File: frontend/src/types/Role.ts
export class Role {
    static ADMIN = new Role('ADMIN');
    static USER = new Role('USER');
    static MANAGER = new Role('MANAGER');
    static VIEWER = new Role('VIEWER');

    private constructor(public readonly name: string) {}

    toString() {
        return this.name;
    }

    static values(): Role[] {
        return [Role.ADMIN, Role.USER, Role.MANAGER, Role.VIEWER];
    }
}

File: backend/src/main/java/com/example/usermanagementbackend/model/Role.java
package com.example.usermanagementbackend.model;

import jakarta.persistence.*;
import java.util.Set;

@Entity
@Table(name = "roles")
public class Role {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true, nullable = false)
    private String name; // e.g., ADMIN, USER

    @ManyToMany(fetch = FetchType.EAGER)
    @JoinTable(
        name = "role_privileges",
        joinColumns = @JoinColumn(name = "role_id"),
        inverseJoinColumns = @JoinColumn(name = "privilege_id")
    )
    private Set<Privilege> privileges;

    // Getters, setters, constructors omitted for brevity
}